packages/components/icon/src/types.ts

```diff
@@ -0,0 +1,18 @@
+import type { PolymorphicComponentProps } from '@contentful/f36-core';
+
+export type IconProps = PolymorphicComponentProps<
+  'svg',
+  {
+    children?: never;
+    color?:
+      | 'muted'
+      | 'negative'
+      | 'positive'
+      | 'primary'
+      | 'secondary'
+      | 'warning'
+      | 'white';
+    icon: string;
+    size?: 'small' | 'large' | 'medium' | 'tiny';
```

Comment:
@denkristoffer Just a call out for later, not really related to this PR. 
Something to be clarified with designers, I guess, is `size` and `color.`
`size` - I think we might need to agree about the sizes that would be shared (maybe partially with IconButton / Button - @mshaaban0 made a proposal about merging those 2)
`color` (maybe it should be `variant`?) - we might want to deprecate muted here as well.

Reply:
Yes, good point thanks, we should go with `variant`.